Add upload file data lambda

MancunianSam opened the pull request:
This is the replacement for tdr-download files. It recieves this `{"consignmentId": "xxxx-xxxx-xxxx", "userId": "xxxx-xxxx-xxxx"}` as input and then:
* Calls the API to get a list of fileIds and original path data
* Gets the list of files from S3
* Compares the two and throws an error if there is a mismatch
* Returns `[{"fileId": "xxxx-xxxx-xxxx" , "originalPath": "/original/file/path", "userId": "xxxx-xxxx-xxxx"}]`

This array will then be passed to the map function in the step function which will call each of the backend checks in turn.

TomJKing reviewed (commented):
Looks good. Thanks. Couple of overall comments:

- Sure its just an oversight, but the README will need updating with information about the repo;
- Missing the `Automatically delete head branches` option in the repo settings
- Will need to add environments for staging/prod

MancunianSam commented:
I've updated the README and I've changed the branch protections. The terraform to do the environments is in, it just needs running on the staging/prod.

TomJKing reviewed (approved):
Looks good. Thanks
